src/web/services/search.service.ts

```diff
@@ -163,14 +163,24 @@ export class SearchService {
 
     let masqueradeGoogleId: string = '';
     for (const instructor of instructors.instructors) {
-      const instructorPrivilege: InstructorPrivilege | undefined = instructorPrivileges.shift();
-      if (instructor.googleId != null
-          && (instructorPrivilege != null && instructorPrivilege.privileges.courseLevel.canModifyInstructor
-              || instructor.role === InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER)) {
+      if (instructor.googleId
+          && instructor.role === InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER) {
         masqueradeGoogleId = instructor.googleId;
         break;
       }
     }
+    // no instructor with co-owner privileges
+    // there is usually at least one instructor with "modify instructor" permission
+    if (masqueradeGoogleId === '') {
+      for (const instructor of instructors.instructors) {
+        const instructorPrivilege: InstructorPrivilege | undefined = instructorPrivileges.shift();
+        if (instructor.googleId
+            && (instructorPrivilege && instructorPrivilege.privileges.courseLevel.canModifyInstructor)) {
```

Comment:
```suggestion
            && instructorPrivilege && instructorPrivilege.privileges.courseLevel.canModifyInstructor) {
```

Reply:
I know the brackets are redundant here, just want to make things clearer